File: app/MainActivity.java
package com.example.clickerapp;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private int counter = 0;
    private boolean increment = true;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView counterTextView = findViewById(R.id.counter);
        Button incrementButton = findViewById(R.id.increment_button);
        Button resetButton = findViewById(R.id.reset_button);
        Button toggleButton = findViewById(R.id.toggle_button);

        incrementButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                counter = increment ? counter + 1 : counter - 1;
                counterTextView.setText(String.valueOf(counter));
            }
        });

        resetButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                counter = 0;
                counterTextView.setText(String.valueOf(counter));
            }
        });

        toggleButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                increment = !increment;
                toggleButton.setText(increment ? "Zmień na dekrementację" : "Zmień na inkrementację");
            }
        });
    }
}

File: app/src/main/java/MainActivity.kt
package com.example.clickerapp

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private var counter = 0
    private var increment = true

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val counterTextView: TextView = findViewById(R.id.counter)
        val incrementButton: Button = findViewById(R.id.increment_button)
        val resetButton: Button = findViewById(R.id.reset_button)
        val toggleButton: Button = findViewById(R.id.toggle_button)

        incrementButton.setOnClickListener {
            counter = if (increment) counter + 1 else counter - 1
            counterTextView.text = counter.toString()
        }

        resetButton.setOnClickListener {
            counter = 0
            counterTextView.text = counter.toString()
        }

        toggleButton.setOnClickListener {
            increment = !increment
            toggleButton.text = if (increment) "Zmień na dekrementację" else "Zmień na inkrementację"
        }
    }
}
